### ISSUE TITLE
no-unused-variable can not detect unused constant

### ISSUE THREAD

**Issue by hpmax00** [palantir/tslint#4077]
### Bug Report

- __TSLint version__:  5.11.0
- __TypeScript version__: 2.9.2 
- __Running TSLint via__: Idea

#### TypeScript code being linted

```ts
const body = {}
```

with `tslint.json` configuration:

```json
    "no-unused-variable": true,
```

#### Actual behavior
<img width="188" alt="image" src="https://user-images.githubusercontent.com/40781942/43375270-95710f08-93e6-11e8-9720-7bfad0c0d0b8.png">

#### Expected behavior
get unused mention

**Comment by JoshuaKGoldberg** [comment]
@hpmax00 this isn't really enough information to go off of ☹ could you please provide the rest of the file? On its own, this isn't actionable.

Closing for now, since `no-unused-variable` directly retrieves diagnostics from TypeScript. We can always re-open if there is something different TSLint can be doing.
